src/deluge/gui/views/performance_session_view.cpp

```diff
@@ -881,18 +897,40 @@ ActionResult PerformanceSessionView::padAction(int32_t xDisplay, int32_t yDispla
 			}
 			uiNeedsRendering(this); // re-render pads
 		}
+		// pressing the first column in sidebar to trigger sections / clips
+		else if (xDisplay == kDisplayWidth) {
```

Comment:
this section would be a lot easier to follow if you inverted the order of the checks - something like this:

if (xDisplay >= kDisplayWidth) {
if (in arranger) {
    do arranger things
}
else {
    if (not in grid) {
        do session view things
    }
    else {
        do grid session things
   }
}
}